waiter/src/waiter/settings.clj

```diff
@@ -214,22 +214,29 @@
 (defn sanitize-settings
   "Sanitizes settings for eventual conversion to JSON"
   [settings]
-  (->> (utils/dissoc-in settings [:zookeeper :connect-string])
-       (walk/postwalk (fn [data]
-                        (if-not (and (map? data) (contains? data :kind))
-                          data
-                          (->> (keys data)
-                               (remove #(let [nested-data (get data %)]
-                                          (and (not= % :kind)
-                                               (not= % (:kind data))
-                                               (map? nested-data)
-                                               (contains? nested-data :factory-fn))))
-                               (select-keys data)))))))
+  (walk/postwalk
+    (fn [data]
+      (if-not (and (map? data) (contains? data :kind))
+        data
+        (->> (keys data)
+             (remove #(let [nested-data (get data %)]
+                        (and (not= % :kind)
+                             (not= % (:kind data))
+                             (map? nested-data)
+                             (contains? nested-data :factory-fn))))
+             (select-keys data))))
+    (cond-> (utils/dissoc-in settings [:zookeeper :connect-string])
```

Comment:
Why are we `dissoc`ing some and using `_hidden_` with others?

Reply:
Done. Use `<hidden>` consistently.